assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External imports
+ */
+import { useRef, useMemo, useCallback } from '@wordpress/element';
+import { reduce } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+
+// @todo, when the wp.data.useSelect hook is available, we can utilize that in
+// this custom hook and then we'll be able to remove the withSelect hoc usage
+// in the composed `TicketAssignmentsManagerModal` component.
+
+const useCountsManager = (
```

Comment:
I'm not sure on the name of this, so I'm open to suggestions for improving the name.